cleaner.go

```diff
@@ -70,79 +76,151 @@ func (c *Cleaner) RemoveServiceCatalogRelease() error {
 	log.Printf("Found %s release in the namespace %s: status %s", release.Name, release.Namespace, release.Info.Status.String())
 	log.Println(" Uninstalling...")
 	err = helmCli.UninstallRelease(&helmclient.ChartSpec{
-		ReleaseName: serviceCatalogReleaseName,
+		ReleaseName: releaseName,
 		Timeout:     time.Minute,
 		Wait:        true,
 	})
 	if err != nil {
 		return err
 	}
+
 	log.Println("DONE")
 	return nil
 }
 
 func (c *Cleaner) RemoveResources() error {
-	err := c.k8sCli.DeleteAllOf(context.Background(), &v1beta1.ServiceInstance{}, client.InNamespace("default"))
-	if err != nil {
-		return err
+	gvkList := []schema.GroupVersionKind{
+		{
+			Kind:    "ServiceBindingUsage",
+			Group:   "servicecatalog.kyma-project.io",
+			Version: "v1alpha1",
+		},
+		{
+			Kind:    "ServiceBinding",
+			Group:   "servicecatalog.k8s.io",
+			Version: "v1beta1",
+		},
+		{
+			Kind:    "ServiceInstance",
+			Group:   "servicecatalog.k8s.io",
+			Version: "v1beta1",
+		},
+		{
+			Kind:    "ServiceBroker",
+			Group:   "servicecatalog.kyma-project.io",
+			Version: "v1alpha1",
+		},
+		{
+			Kind:    "ClusterServiceBroker",
+			Group:   "servicecatalog.kyma-project.io",
+			Version: "v1alpha1",
+		},
 	}
 
-	err = c.k8sCli.DeleteAllOf(context.Background(), &v1beta1.ServiceBinding{}, client.InNamespace("default"))
+	namespaces := &v1.NamespaceList{}
+	err := c.k8sCli.List(context.Background(), namespaces)
 	if err != nil {
 		return err
 	}
 
+	for _, gvk := range gvkList {
+		for _, namespace := range namespaces.Items {
+			log.Printf("%ss in %s\n", gvk.Kind, namespace.Name)
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+			err := c.k8sCli.DeleteAllOf(context.Background(), u, client.InNamespace(namespace.Name))
```

Comment:
for `ClusterServiceBroker` we can probably call the delete just once without providing any namespace given it is a cluster scoped resource. Maybe consider adding a flag to gvkList to distinguish between namespaced and clusterwide?

Reply:
moved outside of the loop (namespaces)